documentation/docs/05-load.md

```diff
@@ -2,253 +2,282 @@
+Both shared and server-only `load` functions have access to properties describing the request (`params`, `routeId` and `url`) and various functions (`depends`, `fetch` and `parent`). These are described in the following sections.
 
-The following example shows how to use `depends` to register a dependency on the URLs to a custom API client as well as a custom identifier, which is `invalidate`d after a button click, making the `load` function rerun.
+Server-only `load` functions are called with a `ServerLoadEvent`, which inherits `clientAddress`, `cookies`, `locals`, `platform` and `request` from `RequestEvent`.
 
-```js
-/// file: src/routes/+page.js
-// @filename: ambient.d.ts
-declare module '$lib/api' {
-	interface Data{}
-	export const base: string;
-	export const client: {
-		get: (resource:string) => Promise<Data>
-	}
-}
+Shared `load` functions are called with a `LoadEvent`, which has a `data` property. If you have `load` functions in both `+page.js` and `+page.server.js` (or `+layout.js` and `+layout.server.js`), the return value of the server-only `load` function is the `data` property of the shared `load` function's argument.
 
-// @filename: index.js
-// ---cut---
-import * as api from '$lib/api';
+#### Output
 
-/** @type {import('./$types').PageLoad} */
-export async function load({ depends }) {
-	depends(
-		`${api.base}/foo`,
-		`${api.base}/bar`,
-		'my-stuff:foo'
-	);
+A shared `load` function can return an object containing any values, including things like custom classes and component constructors.
 
-	return {
-		foo: api.client.get('/foo'),
-		bar: api.client.get('/bar')
-	};
-}
-```
+A server-only `load` function must return data that can be serialized with [devalue](https://github.com/rich-harris/devalue) — anything that can be represented as JSON plus things like `BigInt`, `Date`, `Map`, `Set` and `RegExp`, or repeated/cyclical references — so that it can be transported over the network.
 
-```svelte
-/// file: src/routes/+page.svelte
-<script>
-	import { invalidate } from '$app/navigation';
+#### When to use which
 
-	/** @type {import('./$types').PageData} */
-	export let data;
+Server-only `load` functions are convenient when you need to access data directly from a database or filesystem, or need to use private environment variables.
 
-	const pageRefresh = async () => {
-		await invalidate('my-stuff:foo');
-	}
-</script>
+Shared `load` functions are useful when you need to `fetch` data from an external API and don't need private credentials, since SvelteKit can get the data directly from the API rather than going via your server. They are also useful when you need to return something that can't be serialized, such as a Svelte component constructor.
 
-<p>{data.foo}<p>
-<p>{data.bar}</p>
-<button on:click={pageRefresh}>Refresh my stuff</button>
-```
+In rare cases, you might need to use both together — for example, you might need to return an instance of a custom class that was initialised with data from your server.
```

Comment:
Wondering if this would be a more real-world use case that may also help people discover how to do this:
```suggestion
In rare cases, you might need to use both together — for example, you might need to forward an authentication token from the server for subsequent use in shared `load` functions.
```

Reply:
I don't think we'd want to encourage that pattern. If you have to go the server for an auth token _anyway_ then it makes sense to use it on the server as well — you could get all the data you need in a single hop rather than going to the server for the token then making fetches to the external API from the browser.

The thinking here was that it directly follows from the previous two paragraphs:

1. use server-only load to get stuff from your db
2. use shared load to return funky stuff
3. use both together to return funky stuff initialised with data from the db

Reply:
Let's keep it as-is then